server/src/main/scala/com/cloudera/livy/server/LivyServer.scala

```diff
@@ -90,6 +90,7 @@ class LivyServer extends Logging {
 
     // Initialize YarnClient ASAP to save time.
     if (livyConf.isRunningOnYarn()) {
+      SparkYarnApp.init(livyConf)
```

Comment:
It is not necessary to have a `init(...)` function to create and start the GC thread. We could directly initialize the values inside the `SparkYarnApp`. This values will be initialized when `SparkYarnApp` class is loaded, which is here calling `SparkYarnApp.yarnClient`.